controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java

```diff
@@ -113,13 +114,72 @@ public void deconstruct() {
                          .distinct()
                          .sorted()
-                         .map(cloud -> new OsUpgrader(controller, Duration.ofMinutes(1), cloud))
+                         .map(cloud -> new OsUpgrader(controller, interval, cloud))
                          .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
     }
 
+    private static class Intervals {
+
+        private static final Duration MAX_CD_INTERVAL = Duration.ofHours(1);
+
+        private final SystemName system;
+
+        private final Duration defaultInterval;
+        private final Duration outstandingChangeDeployer;
+        private final Duration versionStatusUpdater;
+        private final Duration readyJobsTrigger;
+        private final Duration deploymentMetricsMaintainer;
+        private final Duration applicationOwnershipConfirmer;
+        private final Duration systemUpgrader;
+        private final Duration jobRunner;
+        private final Duration osUpgrader;
+        private final Duration contactInformationMaintainer;
+        private final Duration nameServiceDispatcher;
+        private final Duration costReportMaintainer;
+        private final Duration resourceMeterMaintainer;
+        private final Duration cloudEventReporter;
+        private final Duration resourceTagMaintainer;
+        private final Duration systemRoutingPolicyMaintainer;
+        private final Duration applicationMetaDataGarbageCollector;
+        private final Duration hostRepairMaintainer;
+        private final Duration containerImageExpirer;
+
+        public Intervals(SystemName system) {
+            this.system = Objects.requireNonNull(system);
+            this.defaultInterval = duration(system.isCd() || system == SystemName.dev ? 1 : 5, MINUTES);
+            this.outstandingChangeDeployer = duration(3, MINUTES);
+            this.versionStatusUpdater = duration(3, MINUTES);
+            this.readyJobsTrigger = duration(1, MINUTES);
+            this.deploymentMetricsMaintainer = duration(5, MINUTES);
+            this.applicationOwnershipConfirmer = duration(12, HOURS);
+            this.systemUpgrader = duration(1, MINUTES);
+            this.jobRunner = duration(90, SECONDS);
+            this.osUpgrader = duration(1, MINUTES);
+            this.contactInformationMaintainer = duration(12, HOURS);
+            this.nameServiceDispatcher = duration(10, SECONDS);
+            this.costReportMaintainer = duration(2, HOURS);
+            this.resourceMeterMaintainer = duration(1, MINUTES);
+            this.cloudEventReporter = duration(30, MINUTES);
+            this.resourceTagMaintainer = duration(30, MINUTES);
+            this.systemRoutingPolicyMaintainer = duration(10, MINUTES);
+            this.applicationMetaDataGarbageCollector = duration(12, HOURS);
+            this.hostRepairMaintainer = duration(12, HOURS);
+            this.containerImageExpirer = duration(2, HOURS);
```

Comment:
I'd love custom table formatting to make the durations easily comparable, but I'm probably weird o_O